src/components/PrimaryNav/PrimaryNav.tsx

```diff
@@ -0,0 +1,127 @@
+import { StatusLabel } from "@canonical/react-components";
+import classNames from "classnames";
+import Logo from "components/Logo/Logo";
+import UserMenu from "components/UserMenu/UserMenu";
+import { useMemo } from "react";
+import { useSelector } from "react-redux";
+import { NavLink } from "react-router-dom";
+import { getAppVersion } from "store/general/selectors";
+import {
+  getControllerData,
+  getGroupedModelStatusCounts,
+} from "store/juju/selectors";
+import { Controllers } from "store/juju/types";
+import "./_primary-nav.scss";
+
+const ModelsLink = () => {
+  const { blocked: blockedModels } = useSelector(getGroupedModelStatusCounts);
+  return (
+    <NavLink
+      className={({ isActive }) =>
+        classNames("p-list__link", {
+          "is-selected": isActive,
+        })
+      }
+      to={"/models"}
+    >
+      <i className={`p-icon--models is-light`}></i>
+      <span className="hide-collapsed">Models</span>
+      {blockedModels > 0 && (
+        <span className="entity-count is-negative">{blockedModels}</span>
+      )}
+    </NavLink>
+  );
+};
+
+const ControllersLink = () => {
+  const controllers: Controllers | null = useSelector(getControllerData);
+
+  const controllersUpdateCount = useMemo(() => {
+    if (!controllers) return 0;
+    let count = 0;
+    Object.values(controllers).forEach((controller) => {
+      controller.forEach((controller) => {
+        if ("version" in controller && controller.updateAvailable) {
+          count += 1;
+        }
+      });
+    });
+    return count;
+  }, [controllers]);
+  //   (acc, controller) => {
+  //     if (!("version" in controller)) return acc;
+  //     return (
+  //       acc +
+  //       controller.reduce((acc, controller: Controller) => {
+  //         return acc + (controller.updateAvailable ? 1 : 0);
+  //       }, 0)
+  //     );
+  //   },
+  //   0
+  // );
```

Comment:
Left some code here.

Reply:
This is a really good idea, nice job Huw :smile: